### PR TITLE
added more tests for TestAssertEqualInstruction

### PR DESCRIPTION
**Pull request by Tomi-3-0**
Added more test cases for the TestAssertEqualInstruction function

### CODE REVIEW

File: pkg/vm/vm_test.go [hunk 1]
```diff
@@ -918,6 +918,78 @@ func TestAssertEqualInstruction(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, mem.MemoryValueFromInt(-5), mv)
 	})
+
+	t.Run("multiply", func(t *testing.T){
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:922]: Could you add one for division as well?

File: pkg/vm/vm_test.go [hunk 2]
```diff
@@ -918,6 +918,78 @@ func TestAssertEqualInstruction(t *testing.T) {
...
+		err:= vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Ap)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(50), mv)
+	})
+
+	t.Run("comparison equality", func(t *testing.T){
+		vm:= defaultVirtualMachineWithCode("[ap + 1] = [fp];")
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:938]: There is already a test that does this

File: pkg/vm/vm_test.go [hunk 3]
```diff
@@ -918,6 +918,78 @@ func TestAssertEqualInstruction(t *testing.T) {
...
+		err:= vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Fp)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(5), mv)
+	})
+
+	t.Run("comparison equality 2", func(t *testing.T){
+		vm:= defaultVirtualMachineWithCode("[ap - 1] = [fp];")
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:952]: There is already a test that does this

File: pkg/vm/vm_test.go [hunk 4]
```diff
@@ -918,6 +918,78 @@ func TestAssertEqualInstruction(t *testing.T) {
...
+
+		err:= vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Fp)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(7), mv)
+	})
+
+	t.Run("compare register values", func(t *testing.T){
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:965]: There is already a test that does this

File: pkg/vm/vm_test.go [hunk 5]
```diff
@@ -918,6 +918,78 @@ func TestAssertEqualInstruction(t *testing.T) {
...
+
+		err:= vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Fp)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(3), mv)
+	})
+
+	t.Run("compare register values left to right", func(t *testing.T){
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:979]: There is no need for this test since there is already others

**Review by rodrigo-pino — COMMENTED** [review]
There are tests with considerable overlap. There is no need to test ap or fp at different sides of assert equal since does individual components are already tested in previous tests like `GetDstCell0` and `GetDstCell1`. Remember that what we are looking for here is to verify that all ways of building an equal instruction works as expected.

File: pkg/vm/vm_test.go (hunk 1 above)
Comment (Tomi-3-0) [reply pkg/vm/vm_test.go:922]: Noted

**Comment by rodrigo-pino** [comment]
Closing, follow up at #133

**Pull request closed without merging**